core/app/models/spree/calculator/promotion/distributed_amount.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_dependency 'spree/calculator'
+
+# This is a calculator for line item adjustment actions. It accepts a line item
+# and calculates its weighted adjustment amount based on the value of the
+# preferred amount and the price of the other line items. More expensive line
+# items will receive a greater share of the preferred amount.
```

Comment:
I think that while we're doing this change, it may be worth moving these comments after the `module Spree` definition, which I think should be the proper location for them, and they can be picked by yard when creating the doc.

Reply:
Done!